packages/apple-pay-integration/e2e/ApplePaySessionCustomerStepMockObject.ts

```diff
@@ -0,0 +1,111 @@
+        }
+
+        static canMakePaymentsWithActiveCard() {
+            return Promise.resolve(this.canMakePayments());
+        }
+
+        completePayment() {
+        }
+
+        completeShippingContactSelection(update) {
+        }
+
+        completeShippingMethodSelection(update) {
+        }
+
+        completeMerchantValidation(response) {
+        }
+        begin() {
+            setTimeout(() => {
+                const validationEvent = {
+                    validationURL: 'https://www.example.com',
+                } as ApplePayJS.ApplePayValidateMerchantEvent
+
+                const shippingContactEvent = {
+                    shippingContact: {
+                        email: "xx@xx.com",
+                        familyName: 'xx',
+                        givenName: 'xx'
+                    }
+                } as unknown as ApplePayJS.ApplePayShippingContactSelectedEvent;
+
+                const shippingMethodSelectedEvent = {
+                    shippingMethod: {
+                        label: 'xx',
+                        detail: 'xx',
+                        amount: "xx",
+                        identifier: 'xx'
+                    }
+                } as ApplePayJS.ApplePayShippingMethodSelectedEvent
+
+                this.onvalidatemerchant(validationEvent);
+                this.onshippingcontactselected(shippingContactEvent);
+                this.onshippingmethodselected(shippingMethodSelectedEvent);
+            }, 0);
+            setTimeout(() => {
+                const mockPaymentData = {
+                    version: "mock_v1",
+                    data: "mockData",
+                    signature: "mockSignature",
+                    header: {
+                        ephemeralPublicKey: "mockPublicKey",
+                        publicKeyHash: "mockPublicKeyHash",
+                        transactionId: "mockTransactionId",
+                    }
+                }
+                const event = {
+                    payment: {
+                        token: {
+                            paymentData: mockPaymentData,
+                            paymentMethod: "xx",
+                            transactionIdentifier: "xx",
+                        },
+                        billingContact: {
+                            emailAddress: "mock@mock.com",
+                            familyName: "mock",
+                            givenName: "mock",
+                            phoneNumber: "00000000",
+                        },
+                        payment: {
+                            shippingContact: {
+                                emailAddress: "mock@mock.com",
+                                familyName: "mock",
+                                givenName: "mock",
+                                phoneNumber: "00000000",
+                            }
+                        },
+                    }
+                } as unknown as ApplePayJS.ApplePayPaymentAuthorizedEvent;
```

Comment:
Is there a reason why you have to use `as unknown as`? Can you just provide dummy data that fully conforms to the `ApplePayPaymentAuthorizedEvent` interface?

Reply:
We would need to add the Event typescript object that isn't related to the ApplePaySession itself, rather the Event focuses more on the browser event handling. Part of it also means we would need to mock EventTarget.

Reply:
Updated the mock object to have the Event definition